src/cpp/rtps/participant/RTPSParticipantImpl.h

```diff
@@ -293,6 +293,11 @@ class RTPSParticipantImpl
                 destination_locators_begin,
                 destination_locators_end,
                 msg->length);
+
+            // checkout if sender is a pdp endpoint
+            on_pdp_packet(sender_guid);
+            // checkout if sender is a edp endpoint
+            on_edp_packet(sender_guid);
```

Comment:
One packet should be accounted per destination. I would follow an approach similar to the one in on_rtps_send(), passing the locator iterators and incrementing the count once per iterator.